### PR TITLE
FO: Add page number in meta titles for new-products, best-sales, prices-drop

### LINKED ISSUE #27596: Page number is not included in meta titles for new-products, best-sales, prices-drop
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

As the title says, page number is not included in meta_title of some product listing controllers. 

### Expected behavior

If page number is > 1, it should be included in meta_title for better SEO. 
For example, when user is on 2nd page of Best Sellers page, meta title should be "Best Sales (2)". 
Currently page number is added only on Category and manufacturer listing pages. 

### Steps to reproduce

1. Go to 2nd page of 'newproducts' or 'bestsales' or 'pricesdrop'

### PrestaShop version(s) where the bug happened

all

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Amazzing**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add page number in **meta titles** for listing pages that have pagination, like `new-products`, `best-sales`, `prices-drop`, etc.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/27596
| Related PRs       | 
| How to test?      | Go to 2nd page of any controller with pagination and check **meta title**
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28297)

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

File: classes/Meta.php [hunk 1]
```diff
@@ -528,9 +519,30 @@ public static function completeMetaTags($metaTags, $defaultValue, Context $conte
...
+     *
+     * @param string $title
+     *
+     * @return string
+     */
+    public static function paginateTitle(string $title): string
+    {
+        $page_num = (int) Tools::getValue('page');
+        if ($page_num > 1) {
+            $title .= ' (' . $page_num . ')';
```
Comment (matthieu-rolland) [inline classes/Meta.php:540]:
```suggestion
            $title .= sprintf(' (%s)', $page_num);
```

It's usually better to format strings with sprintf rather than using concatenation (but I might be nitpicking here)

**Review by matthieu-rolland — COMMENTED** [review]
thank you for your contribution @Amazzing 👍 

One small feedback, but LGTM otherwise

File: classes/Meta.php (hunk 1 above)
Comment (Amazzing) [reply classes/Meta.php:540]: Hello, @matthieu-rolland
`sprintf` could be really useful if that was a translatable/editable string. But it is not translatable/editable. So, are you sure there are any advantages of using `sprintf` in this particular case?

**Review by kpodemski — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @Amazzing ,

Thank you for your PR, I tested it and that what I see : 
image

If it's not the good way to test it, can you be more explicit on you how to test ? I'll try it quickly after your answer

**Comment by kpodemski** [comment]
Hi @AureRita 

Is this a second page of the results? You need to set, for instance, 3 products per page and make sure that when you switch the page, you have a different meta title

**Comment by AureRita** [comment]
Hi @kpodemski ,

Yes, i'm currently on the second page of the result, I'll set 3 product per page, it's maybe because 2 pages is not enought

**Comment by kpodemski** [comment]
@AureRita, if you are on the second page and there's no indication about it in `meta-title`, then something is wrong with the PR

I think @Amazzing had not taken into consideration that there's no condition to display a dedicated meta title for the New products page :-)

Could you check this @Amazzing ?

**Comment by AureRita** [comment]
Thank you @kpodemski ,

The problem is the number display, as you can see : 
image

I check many times and i see that meta title is refreshed only when you do a hard refresh on the page ( Ctrl + Maj + R )

**Comment by kpodemski** [comment]
Ok @AureRita 

For the search engines, that's ok. But it would be good to alter this behavior for the end-user too. Let's wait for @Amazzing response :)

**Comment by Amazzing** [comment]
Hello, @AureRita,
When you click on a pagination link, the page is not reloaded. Product list is dynamically updated via ajax. Other elements on page including meta title are not updated. It's always been like that in PrestaShop. 
Page number is included in meta title when pagination link is actually loaded (obviously it is included when search engine crawls a pagination link)
Theoretically we could update meta title dynamically for users too. I will check what can be done about it later.

**Comment by kpodemski** [comment]
I think it's related to the `ps_facetetsearch` module @Amazzing, right? If so, it could be done in a separate PR. This one seems to be ok.

**Comment by AureRita** [comment]
That what I thinked, this is updated by ajax, so yes I think It's QA Approved

**Comment by Progi1984** [comment]
Thanks @Amazzing & @AureRita
